gui/src/components/Settings/Wallet/JsonKeystore.tsx

```diff
@@ -50,7 +53,8 @@ export function JsonKeystore({
           color="primary"
           variant="contained"
           type="submit"
-          disabled={!isDirty || !isValid}
+          disabled={(!isDirty || !isValid) && !isActuallyDirty}
+          onClick={() => setIsActuallyDirty(true)}
```

Comment:
I'm not understanding this logic

clicking on the submit button makes the dirty state true? that sounds backwards. in fact, check this flow:

- edit an item (e.g.: the "dev account" checkbox)
- save the wallet, which closes the accordion
- open the wallet again. the save button is now enabled, even though it's in the original state

what I would imagine a correct solution does is:

- each wallet uses `defaultValues: wallet`. which is what's used to compare with current values and compute `isDirty`
- if we have this value updated (which should be happening automatically, but maybe there's some break along the stack I'm missing), this should "just work". Ideally we'd find out where the reactivity is missing and fix there

Reply:
Thinking about this again, maybe its just a matter of calling https://react-hook-form.com/docs/useform/reset ?